### PR TITLE
ECOM-5486 pass sailthru campaign id to Sailthru for enrolls

### PR DESCRIPTION
**Pull request by PaulWattenberger**

### CODE REVIEW

**Comment by PaulWattenberger** [comment]
@clintonb code to retain sailthru cookie when handling audit/honor enrolls with async fulfillment enabled

File: ecommerce/sailthru/signals.py [hunk 1]
```diff
@@ -115,11 +111,15 @@ def process_basket_addition(sender, product=None, user=None, request=None, baske
...
 
-        # pass event to ecommerce_worker.sailthru.v1.tasks to handle asynchronously
-        update_course_enrollment.delay(user.email, _build_course_url(course_id), True, mode_for_seat(product),
-                                       unit_cost=price, course_id=course_id, currency=currency,
-                                       site_code=partner.short_code,
-                                       message_id=message_id)
+        # inform sailthru if there is a price, no need to tell sailthru about adding enrolls to shopping cart since
+        # abandoned cart is not possible
+        if price:
+
```
Comment (clintonb) [inline ecommerce/sailthru/signals.py]: Remove the unnecessary break.

File: ecommerce/sailthru/signals.py [hunk 2]
```diff
@@ -115,11 +111,15 @@ def process_basket_addition(sender, product=None, user=None, request=None, baske
...
+            # pass event to ecommerce_worker.sailthru.v1.tasks to handle asynchronously
```
Comment (clintonb) [inline ecommerce/sailthru/signals.py]: This comment doesn't add any value to the reader, and should be removed.

File: ecommerce/sailthru/signals.py [hunk 3]
```diff
@@ -115,11 +111,15 @@ def process_basket_addition(sender, product=None, user=None, request=None, baske
                 value_text=message_id
             )
 
-        # pass event to ecommerce_worker.sailthru.v1.tasks to handle asynchronously
-        update_course_enrollment.delay(user.email, _build_course_url(course_id), True, mode_for_seat(product),
-                                       unit_cost=price, course_id=course_id, currency=currency,
-                                       site_code=partner.short_code,
-                                       message_id=message_id)
+        # inform sailthru if there is a price, no need to tell sailthru about adding enrolls to shopping cart since
```
Comment (clintonb) [inline ecommerce/sailthru/signals.py]: What does the second half of this comment mean? Why does abandonment matter here? If the user has enrolled, Sailthru should be notified.

**Review by clintonb — APPROVED** [review]

File: ecommerce/sailthru/signals.py (hunk 3 above)
Comment (PaulWattenberger) [reply ecommerce/sailthru/signals.py]: Sailthru needs to be told when enrollment complete, but not when the enroll is added to the basket.  That is only needed for abandoned cart detection...

Comment (clintonb) [reply ecommerce/sailthru/signals.py]: Please update the comment to be clearer:

```
We only need to notify Sailthru when there is the possibility of an abandoned cart. Only paid orders have the possibility of being abandoned, so we only notify Sailthru for paid orders.
```

**Review by clintonb — APPROVED** [review]

**Pull request merged**